src/halmos/__main__.py

```diff
@@ -638,34 +794,17 @@ def main() -> int:
     # command line arguments
     #
 
+    global args
```

Comment:
I don't love introducing globals, but in this case I think it's reasonable because args is written to only once but read by basically function, which means that it pollutes every function parameter list

Reply:
ok, let's improve this later if needed.